### PR TITLE
Add header.build back in as a useful hook

### PR DESCRIPTION
**Pull request by pitaj**
`filter:header.build` can now be used to augment the header template variables. Since the header variables can be accessed by the whole page (I think?) this is a very useful feature.

Deprecation is [here](https://github.com/NodeBB/NodeBB/blob/a355fbfc81f451fab32d94d5e19625bccc259718/src/navigation/admin.js#L57)

### CODE REVIEW

**Comment by pitaj** [comment]
Any comments on this @julianlam @psychobunny @barisusakli

File: src/middleware/middleware.js [hunk 1]
```diff
@@ -174,11 +174,16 @@ middleware.buildHeader = function(req, res, next) {
 			},
 			footer: function(next) {
 				app.render('footer', {loggedIn: (req.user ? parseInt(req.user.uid, 10) !== 0 : false)}, next);
+			},
+			plugins: function(next){
+				plugins.fireHook('filter:header.build', res.locals, next);
```
Comment (barisusakli) [inline src/middleware/middleware.js]: Can you change this to 

`plugins.fireHook('filter:header.build', {locals: res.locals}, next);`

**Comment by pitaj** [comment]
@barisusakli How's it look now?

**Comment by barisusakli** [comment]
@psychobunny not sure if the deprecated hook should be removed. Let me know.

**Comment by pitaj** [comment]
Any updates on this?

The deprecated hook can either stay or go. The functionality isn't compromised either way regarding this PR. The PR code won't affect any code already using this hook.

**Comment by julianlam** [comment]
`header.build` _might not_ be able to be added back in as-is because the name is now in reference to building navigational header items in the ACP.

If you rename the hook, I'd be happier with accepting it.

**Comment by pitaj** [comment]
Well, but that function of it is deprecated, and still works with this PR
in place.

**Comment by pitaj** [comment]
In other words, just merge it already. There is no downside.

**Pull request closed without merging**